snuba/datasets/plans/single_storage.py

```diff
@@ -46,14 +54,16 @@ def __init__(
         self.__post_processors = post_processors or []
 
     def build_plan(self, request: Request) -> StorageQueryPlan:
-        # the Request object is immutable, the query processing still depends on the same
-        # request object to be moved around, thus the Query has to be mutable and cannot
-        # be simply rebuilt and replaced.
-        request.query.set_data_source(
+        # TODO: Clearly the QueryTranslator instance  will be dependent on the storage.
+        # Setting the data_source on the query should become part of the translation
+        # as well.
```

Comment:
The main question here, that I think is besides the point of this PR since both solutions are viable, is whether:
1) the storage will provide a translator interface (which hopefully an instance of a general class that is capable to translate any schema into any schema)
2) We will assign here a translator that we build here starting from the schema of the entity and the schema of the storage
3) the storage just provides a "translate" method (which is probably going to be the case for the multi-table storages. 

I think in any of these cases there is likely going to be a translator interface that takes a query and spits a physical query.

Reply:
This has been discussed offline. The translator will be a visitor.